src/manifolds/Hyperbolic.jl

```diff
@@ -31,12 +31,223 @@ struct Hyperbolic{N} <: AbstractEmbeddedManifold{ℝ,DefaultIsometricEmbeddingTy
 
 Hyperbolic(n::Int) = Hyperbolic{n}()
 
+#
+# Representations
+#
+
+@doc raw"""
+    HyperboloidPoint <: MPoint
+
+In the Hyperboloid model of the [`Hyperbolic`](@ref) $ℍ^n$ points are represented
+as vectors in $ℝ^{n+1}$ with [`MinkowskiMetric`](@ref) equal to $-1$.
+
+This representation is the default, i.e. vectors are assumed to have this repesentation.
+"""
+struct HyperboloidPoint{T} <: MPoint
+    value::Vector{T}
+end
+function convert(::Type{HyperboloidPoint{T1}}, x::Array{T2,1}) where {T1,T2}
+    return HyperboloidPoint{T1}(x)
+end
+convert(::Array{T1,1}, x::HyperboloidPoint{T2}) where {T1,T2} = x.value
+
+@doc raw"""
+    HyperboloidTVector <: TVector
+
+In the Hyperboloid model of the [`Hyperbolic`](@ref) $ℍ^n$ tangent vctors are represented
+as vectors in $ℝ^{n+1}$ with [`MinkowskiMetric`](@ref) $⟨p,X⟩_{\mathrm{M}}=0$ to their base
+point $p$.
+
+This representation is the default, i.e. vectors are assumed to have this repesentation.
+"""
+struct HyperboloidTVector{T} <: MPoint
+    value::Vector{T}
+end
+function convert(::Type{HyperboloidTVector}, x)
+    return HyperboloidTVector(x)
+end
+
+@doc raw"""
+    PoincareBallPoint <: MPoint
+
+A point on the [`Hyperbolic`](@ref) manifold $ℍ^n$ can be represented as a vector of norm
+less than one in $\mathbb R^n$.
+"""
+struct PoincareBallPoint{T} <: MPoint
+    value::Vector{T}
```

Comment:
```suggestion
struct PoincareBallPoint{TValue<:AbstractVector} <: MPoint
    value::TValue
```
So that it can wrap around any vector representation.

Reply:
Cool. I will adapt the others, too.